src/modules/quizzes.js

```diff
@@ -0,0 +1,194 @@
+  const {
+    apiKey,
+    clientId,
+    sessionId,
+    segments,
+    userId,
+    version,
+  } = options;
+  const serviceUrl = 'https://quizzes.cnstrc.com';
+  let queryParams = { c: version };
+  let answersParamString = '';
+
+  queryParams.key = apiKey;
+  queryParams.i = clientId;
+  queryParams.s = sessionId;
+
+  // Pull user segments from options
+  if (segments && segments.length) {
+    queryParams.us = segments;
+  }
+
+  // Pull user id from options
+  if (userId) {
+    queryParams.ui = userId;
+  }
+
+  // Validate quiz id is provided
+  if (!quizId || typeof quizId !== 'string') {
+    throw new Error('quizId is a required parameter of type string');
+  }
+
+  if (path === 'finalize' && (typeof parameters.a !== 'object' || !Array.isArray(parameters.a) || parameters.a.length === 0)) {
+    throw new Error('a is a required parameter of type array');
+  }
+
+  if (parameters) {
+    const { section, a, versionId } = parameters;
+
+    // Pull section from parameters
+    if (section) {
+      queryParams.section = section;
+    }
+
+    // Pull version_id from parameters
+    if (versionId) {
+      queryParams.version_id = versionId;
+    }
+
+    // Pull a from parameters and transform
+    if (a) {
+      a.forEach((ans) => {
+        answersParamString += `&${qs.stringify({ a: ans }, { arrayFormat: 'comma' })}`;
+      });
+    }
+  }
+
+  queryParams._dt = Date.now();
+  queryParams = helpers.cleanParams(queryParams);
+
+  const queryString = qs.stringify(queryParams, { indices: false });
+
+  return `${serviceUrl}/v1/quizzes/${encodeURIComponent(quizId)}/${encodeURIComponent(path)}/?${queryString}${answersParamString}`;
+}
+
+/**
+ * Interface to quiz related API calls
+ *
+ * @module quizzes
+ * @inner
+ * @returns {object}
+ */
+class Quizzes {
+  constructor(options) {
+    this.options = options || {};
+    this.eventDispatcher = new EventDispatcher(options.eventDispatcher);
+  }
+
+  /**
+   * Retrieve next quiz from api
+   *
+   * @function getNextQuiz
+   * @description Retrieve next quiz from Constructor.io API
+   * @param {string} id - The id of the quiz
+   * @param {string} [parameters] - Additional parameters to refine result set
+   * @param {string} [parameters.section] - Section for customer's product catalog (optional)
+   * @param {array} [parameters.a] - An array for answers in the format [[1,2],[1]]
+   * @param {string} [parameters.version_id] - Specific version id for the quiz.
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {Promise}
+   * @see https://quizzes.cnstrc.com/api/#/quizzes/QuizzesController_getNextQuestion
+   * @example
+   * constructorio.search.getNextQuiz('quizId', {
+   *    a: [[1,2],[1]],
+   *    section: "123",
+   *    version_id: "123"
+   * });
+   */
+  getNextQuiz(quizId, parameters, networkParameters = {}) {
```

Comment:
What do you think of changing this to `getNextQuestion`? I think we currently call `constructorio.quizzes.getNextQuiz` which is a little weird since we're not getting the next question, not a quiz

EDIT: I just realized Enes made the same comment above 😂